src/storage/mutate/UpdateEdgeProcessor.cpp

```diff
@@ -268,13 +270,13 @@ std::string UpdateEdgeProcessor::updateAndWriteBack(PartitionID partId,
         std::unique_ptr<RowReader> reader, rReader;
         for (auto& index : indexes_) {
             auto indexId = index->get_index_id();
-            if (index->get_schema_id().get_edge_type() == edgeKey.edge_type) {
+            if (index->get_schema_id().get_edge_type() == std::abs(edgeKey.edge_type)) {
                 if (!val_.empty()) {
                     if (rReader == nullptr) {
                         rReader = RowReader::getEdgePropReader(this->schemaMan_,
                                                                val_,
                                                                spaceId_,
-                                                               edgeKey.edge_type);
+                                                               std::abs(edgeKey.edge_type));
                     }
                     auto rValues = collectIndexValues(rReader.get(),
                                                       index->get_fields());
```

Comment:
Let me handle the in-edge prop of index with insert edge together.
Could you help change back line 273, 279 and 295?

Reply:
Let me check the edge_type, if it is negative, just skip directly.  You could handle it later.

Reply:
> Let me check the edge_type, if it is negative, just skip directly. You could handle it later.

excellent!